### PR TITLE
RoundRobinLoadBalancer minor cleanups

### PR DESCRIPTION
**Pull request by lukaszx0**
A bunch of code clean up in `RoundRobinLoadBalancer2`

/cc @kkaneda  @zhangkun83

### CODE REVIEW

File: core/src/main/java/io/grpc/util/RoundRobinLoadBalancerFactory2.java [hunk 1]
```diff
@@ -241,28 +241,27 @@ private Status getAggregatedError() {
     final Status status;
```
Comment (kkaneda) [inline core/src/main/java/io/grpc/util/RoundRobinLoadBalancerFactory2.java]: add `private` for consistency?

File: core/src/main/java/io/grpc/util/RoundRobinLoadBalancerFactory2.java [hunk 2]
```diff
@@ -241,28 +241,27 @@ private Status getAggregatedError() {
...
     private final List<Subchannel> list;
     private int index;
```
Comment (kkaneda) [inline core/src/main/java/io/grpc/util/RoundRobinLoadBalancerFactory2.java]: maybe mention that `index` is guarded by this?

File: core/src/main/java/io/grpc/util/RoundRobinLoadBalancerFactory2.java (hunk 1 above)
Comment (lukaszx0) [inline core/src/main/java/io/grpc/util/RoundRobinLoadBalancerFactory2.java]: Actually it's used in tests. I've added pkg private getter and made it private

File: core/src/main/java/io/grpc/util/RoundRobinLoadBalancerFactory2.java (hunk 2 above)
Comment (lukaszx0) [inline core/src/main/java/io/grpc/util/RoundRobinLoadBalancerFactory2.java]: Done

File: core/src/main/java/io/grpc/util/RoundRobinLoadBalancerFactory2.java [hunk 3]
```diff
@@ -238,31 +238,31 @@ private Status getAggregatedError() {
...
+      if (status != null) {
+        return PickResult.withError(status);
+      }
+
+      return PickResult.withNoResult();
     }
 
     private Subchannel nextSubchannel() {
-      if (empty) {
+      if (list.isEmpty()) {
```
Comment (zhangkun83) [inline core/src/main/java/io/grpc/util/RoundRobinLoadBalancerFactory2.java]: How about saving `size` as a member variable? This way you can eliminate `isEmpty()` call here and the `list.size()` call on line 271 and 272. In fact line 272 could be `index = 0`.

Comment (lukaszx0) [inline core/src/main/java/io/grpc/util/RoundRobinLoadBalancerFactory2.java]: Good call, fixed. Reshuffled and changed first commit, which is now last (might be worth to merge those fixes as a separate commits, that's why I'm keeping them separate - your call)

Comment (zhangkun83) [inline core/src/main/java/io/grpc/util/RoundRobinLoadBalancerFactory2.java]: They are all very small changes. I am going to squash them.

**Pull request merged**